cfme/markers/env_markers/provider.py

```diff
@@ -158,6 +158,15 @@ class DataProvider(object):
     type_name = attr.ib()
     version = attr.ib()
     klass = attr.ib(default=attr.Factory(data_provider_types, takes_self=True))
+    the_id = attr.ib(init=False)
+
+    def __attrs_post_init__(self):
```

Comment:
Could this just be a DataProvider property instead of a custom init attr?

Reply:
yup, do you think it's necessary ? 
we use attrs. so, I try using its stuff instead.

Reply:
I don't see a need to set it at DataProvider instantiation, which is when I would use this. To me, just a regular property is more straightforward.